common/scala/src/main/scala/org/apache/openwhisk/connector/kafka/KafkaConsumerConnector.scala

```diff
@@ -132,6 +132,11 @@ class KafkaConsumerConnector(
         } else {
           throw e
         }
+      case e: WakeupException =>
+        logging.info(this, s"WakeupException happened when do commit action for topic ${topic}")
+        close()
+        recreateConsumer()
```

Comment:
I think `recreateConsumer()` already includes `consumer.close()`?
https://github.com/apache/incubator-openwhisk/pull/4459/commits/4e9541a16cf26b67cc4e1e5d0a8fcafa0b599d74#diff-8bf7d87d98238d39ba7f36d2c52def6fL171

Reply:
Oh, good catch. fixed

Reply:
`wakeup()` is called by the `peak()` method above. Wouldn't recreating the consumer here result in issues when we call `wakeup()` from the `peak()` method? Ie, we'd recreate the consumer here while `peak()` is still trying to do work.

Reply:
@dubee 
Since we open neither `consumer` field nor `wakeup()` method, and currently `wakeup()` method is only called by `wakeupTask` scheduler, I think there would be no case that `WakeupException` thrown by `wakeupTask` is caught here.

In the future, if we expose `wakeup()` method, while `commit()` is called, `wakeup()` method can be called in a different thread, then this change would help to handle it.
Since all accesses to `consumer` are synchronized by `synchronized` keyword, it would be guaranteed that `wakeup()` would "wake up" only one thread at a time.